code/modules/projectiles/guns/ballistic.dm

```diff
@@ -181,7 +181,7 @@
 		. += "[icon_state]_toy"
 
 
-	if(!magazine || internal_magazine)
+	if(!magazine || internal_magazine || !mag_display)
```

Comment:
honestly i still think removing this var and adding an invisible `flatgun_mag` icon is a far better option over adding a snowflake check that's run every single time a ballistic gun fires just for a single meme spaceruin weapon.

Reply:
my concern is more stuff that doesn't want to display mags popping up, and then people duplicate the solution, or worse, add the var themselves and don't know to update the flatgun

Reply:
If you just want it gone I can do that